From a1f82d619eb4c9dea77e7c84b4dbff96325925bd Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Wed, 18 May 2016 13:39:11 +0300
Subject: [PATCH 1144/1240] gpio: dts: a8k: add GPIO nodes in device-tree of
 Armada-70x0/80x0

Add GPIO nodes in device-tree of Armada-70x0/80x0, and add simple-bus entries for UCLASS.

Change-Id: Ibe2ee9a9ed881afe2da4d695d9e0ccec160fb32c
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/29793
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Haim Boot <hayim@marvell.com>
---
 arch/arm/dts/apn-806-a0.dtsi     | 11 +++++++++--
 arch/arm/dts/apn-806-pd-a0.dts   |  1 +
 arch/arm/dts/armada-70x0-rz.dtsi |  1 +
 arch/arm/dts/armada-70x0.dtsi    |  1 +
 arch/arm/dts/armada-80x0.dtsi    |  1 +
 arch/arm/dts/cp110-master.dtsi   |  9 +++++++--
 arch/arm/dts/cp110.dtsi          | 16 ++++++++++++++++
 7 files changed, 36 insertions(+), 4 deletions(-)

diff --git a/arch/arm/dts/apn-806-a0.dtsi b/arch/arm/dts/apn-806-a0.dtsi
index 97b6c90..7c08668 100644
--- a/arch/arm/dts/apn-806-a0.dtsi
+++ b/arch/arm/dts/apn-806-a0.dtsi
@@ -32,12 +32,12 @@
 	};
 
 	ap-806 {
-		compatible = "marvell,mvebu-apn-806";
+		compatible = "marvell,mvebu-apn-806", "simple-bus";
 		#address-cells = <1>;
 		#size-cells = <0>;
 
 		internal-regs {
-			compatible = "marvell,internal-regs";
+			compatible = "marvell,internal-regs", "simple-bus";
 			#address-cells = <1>;
 			#size-cells = <1>;
 			ranges = <0x0000 0xf0000000 0x1000000>;
@@ -73,6 +73,13 @@
 				pin-count = <20>;
 				max-func = <3>;
 			};
+			gpio0: gpio@6F5040 {
+				compatible = "marvell,mvebu-gpio";
+				reg = <0x6F5040 0x40>;
+				bank-name = "armada-ap806";
+				ngpios = <ARMADA_AP806_GPIO_COUNT>;
+				status = "okay";
+			};
 
 			ccu {
 				compatible = "marvell,mvebu-ccu";
diff --git a/arch/arm/dts/apn-806-pd-a0.dts b/arch/arm/dts/apn-806-pd-a0.dts
index fb01833..e192a3a 100644
--- a/arch/arm/dts/apn-806-pd-a0.dts
+++ b/arch/arm/dts/apn-806-pd-a0.dts
@@ -18,6 +18,7 @@
 
 /dts-v1/;
 
+#include <dt-bindings/gpio/armada-8k-gpio.h>
 #include "apn-806-a0.dtsi"
 
 / {
diff --git a/arch/arm/dts/armada-70x0-rz.dtsi b/arch/arm/dts/armada-70x0-rz.dtsi
index 87cfab5..0949557 100644
--- a/arch/arm/dts/armada-70x0-rz.dtsi
+++ b/arch/arm/dts/armada-70x0-rz.dtsi
@@ -16,6 +16,7 @@
  * ***************************************************************************
  */
 
+#include <dt-bindings/gpio/armada-8k-gpio.h>
 #include "apn-806-z1.dtsi"
 #include "cp110-master.dtsi"
 
diff --git a/arch/arm/dts/armada-70x0.dtsi b/arch/arm/dts/armada-70x0.dtsi
index 41ff83f..fa6c48c 100644
--- a/arch/arm/dts/armada-70x0.dtsi
+++ b/arch/arm/dts/armada-70x0.dtsi
@@ -18,6 +18,7 @@
 
 #include <dt-bindings/ddr/ddr_data.h>
 #include <dt-bindings/comphy/comphy_data.h>
+#include <dt-bindings/gpio/armada-8k-gpio.h>
 #include "apn-806-a0.dtsi"
 #include "cp110-master.dtsi"
 
diff --git a/arch/arm/dts/armada-80x0.dtsi b/arch/arm/dts/armada-80x0.dtsi
index 3b1efb9..c3f5079 100644
--- a/arch/arm/dts/armada-80x0.dtsi
+++ b/arch/arm/dts/armada-80x0.dtsi
@@ -17,6 +17,7 @@
  */
 
 #include <dt-bindings/comphy/comphy_data.h>
+#include <dt-bindings/gpio/armada-8k-gpio.h>
 #include "apn-806-a0.dtsi"
 #include "cp110-master.dtsi"
 #include "cp110-slave.dtsi"
diff --git a/arch/arm/dts/cp110-master.dtsi b/arch/arm/dts/cp110-master.dtsi
index 7ba5101..6c860ce 100644
--- a/arch/arm/dts/cp110-master.dtsi
+++ b/arch/arm/dts/cp110-master.dtsi
@@ -21,12 +21,12 @@
 	model = "Marvell Armada CPN-110 Master";
 
 	cp110-master {
-		compatible = "marvell,armada-cpn110-master";
+		compatible = "marvell,armada-cpn110-master", "simple-bus";
 		#address-cells = <1>;
 		#size-cells = <1>;
 
 		internal-regs {
-			compatible = "marvell,internal-regs";
+			compatible = "marvell,internal-regs", "simple-bus";
 			ranges = <0x0000 0xf2000000 0x1000000>;
 			#address-cells = <1>;
 			#size-cells = <1>;
@@ -42,6 +42,11 @@
 			sar-reg {
 				status = "okay";
 			};
+			gpio1: gpio@440100 {
+			};
+			gpio2: gpio@440140 {
+			};
+
 			gop {
 				emac0: mac0 {
 					mac-address = [00 00 00 00 00 01];
diff --git a/arch/arm/dts/cp110.dtsi b/arch/arm/dts/cp110.dtsi
index 4b5e921..a10472e 100644
--- a/arch/arm/dts/cp110.dtsi
+++ b/arch/arm/dts/cp110.dtsi
@@ -120,6 +120,22 @@ pinctl@440000 {
 	max-func = <0xf>;
 };
 
+gpio@440100 {
+	compatible = "marvell,mvebu-gpio";
+	reg = <0x440100 0x40>;
+	bank-name = "armada-cp0";
+	ngpios = <ARMADA_CP0_B0_GPIO_COUNT>;
+	status = "okay";
+};
+
+gpio@440140 {
+	compatible = "marvell,mvebu-gpio";
+	reg = <0x440140 0x40>;
+	bank-name = "armada-cp1";
+	ngpios = <ARMADA_CP0_B1_GPIO_COUNT>;
+	status = "okay";
+};
+
 nand@720000 {
 	compatible = "marvell,mvebu-pxa3xx-nand";
 	reg = <0x720000 0x100>;
-- 
1.9.1

